models/zeroex/bnb/zeroex_bnb_api_fills.sql

```diff
@@ -0,0 +1,448 @@
+            block_time                                      AS block_time,
+            '0x' || substring(DATA, 27, 40)                 AS maker,
+            '0xdef1c0ded9bec7f1a1670819833240f027b25eff'    AS taker,
+            '0x' || substring(DATA, 91, 40)                 AS taker_token,
+            '0x' || substring(DATA, 155, 40)                AS maker_token,
+            bytea2numeric_v3('0x' || substring(DATA, 219, 40)) AS taker_token_amount_raw,
+            bytea2numeric_v3('0x' || substring(DATA, 283, 40)) AS maker_token_amount_raw,
+            'NewBridgeFill'                                 AS type,
+            zeroex_tx.affiliate_address                     AS affiliate_address,
+            TRUE                                            AS swap_flag,
+            FALSE                                           AS matcha_limit_order_flag
+    FROM {{ source('bnb' ,'logs') }} logs
+    JOIN zeroex_tx ON zeroex_tx.tx_hash = logs.tx_hash
+    WHERE topic1 = '0xe59e71a14fe90157eedc866c4f8c767d3943d6b6b2e8cd64dddcc92ab4c55af8'
+        AND contract_address = '0xdb6f1920a889355780af7570773609bd8cb1f498'
+
+        {% if is_incremental() %}
+        AND block_time >= date_trunc('day', now() - interval '1 week')
+        {% endif %}
+        {% if not is_incremental() %}
+        AND block_time >= '{{zeroex_v4_start_date}}'
+        {% endif %}
+),
+direct_PLP AS (
+    SELECT
+            plp.evt_tx_hash,
+            plp.evt_index               AS evt_index,
+            plp.contract_address,
+            plp.evt_block_time          AS block_time,
+            provider                    AS maker,
+            recipient                   AS taker,
+            inputToken                  AS taker_token,
+            outputToken                 AS maker_token,
+            inputTokenAmount            AS taker_token_amount_raw,
+            outputTokenAmount           AS maker_token_amount_raw,
+            'LiquidityProviderSwap'     AS type,
+            zeroex_tx.affiliate_address AS affiliate_address,
+            TRUE                        AS swap_flag,
+            FALSE                       AS matcha_limit_order_flag
+    FROM {{ source('zeroex_bnb', 'ExchangeProxy_evt_LiquidityProviderSwap') }} plp
+    JOIN zeroex_tx ON zeroex_tx.tx_hash = plp.evt_tx_hash
+
+    {% if is_incremental() %}
+    WHERE evt_block_time >= date_trunc('day', now() - interval '1 week')
+    {% endif %}
+    {% if not is_incremental() %}
+    WHERE evt_block_time >= '{{zeroex_v3_start_date}}'
+    {% endif %}
+
+),
+uni_v2_swap as (
+SELECT   s.tx_hash tx_hash, s.index evt_index, s.contract_address, s.block_time, 
+    '0x' || substring(DATA, 27, 40) AS maker, 
+    '0xdef1c0ded9bec7f1a1670819833240f027b25eff' AS taker,
+                '0x' || substring(DATA, 91, 40) AS taker_token,
+            '0x' || substring(DATA, 155, 40) AS maker_token,
+            bytea2numeric('0x' || substring(DATA, 219, 40)) AS taker_token_amount_raw,
```

Comment:
do you want to use v3 function here, like [above](https://github.com/duneanalytics/spellbook/pull/3081/files#diff-da823930ae34f283e8537ff1d497676c1d7bdf32a0069fc643a262222b02ec5bR232)?

Reply:
nice catch